src/components/form/form.types.ts

```diff
@@ -36,6 +36,13 @@ export interface FormError {
     message: string;
 }
 
+export type ValidationError = {
+    /**
+     * Name of the field the error belongs to
+     */
+    [key: string]: string[] | ValidationError;
+};
```

Comment:
I was thinking a bit about both the name of this and what the best structure would be, `FormError` is already taken and has a structure that is not optimal for this use case. It would have been nice if we could have the same format both here and where `FormError` is used right now, but it will probably cause more work for both consumers and ourselves since we have to convert the errors between different formats.

The format here is very close to what Rjsf wants and is also the same format that Marshmallow will return when there is a validation error so I think this will be a good middle ground.